src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -122,12 +122,34 @@ export function toAbsoluteUri(relativeUri: string) {
   return testAnchor.href;
 }
 
-function findClosestAncestor(element: Element | null, tagName: string) {
+function findAnchorTaget(event: MouseEvent): HTMLAnchorElement | null {
+  const path = event.composedPath && event.composedPath();
+  if (path) {
+    // This logic works with events that target elements within a shadow root,
+    // as long as the shadow mode is 'open'. For closed shadows, we can't possibly
+    // know what internal element was clicked.
+    for (let i = 0; i < path.length; i++) {
+      const candidate = path[i];
+      if (candidate instanceof Element && candidate.tagName === 'A') {
+        return candidate as HTMLAnchorElement;
+      }
+    }
+    return null;
+  } else {
+    // Since we're adding use of composedPath in a patch, retain compatibility with any
+    // legacy browsers that don't support it by falling back on the older logic, even
+    // though it won't work properly with ShadowDOM. This can be removed in the next
+    // major release.
```

Comment:
AFAIK there aren't any supported browsers that don't support Event.composedPath.

The only reason I'm keeping the older logic as fallback is in case someone has somehow polyfilled enough to make Blazor Server 5.0 run on IE11, in which case we don't want this patch to create the need to add yet another polyfill. I doubt anyone is in that situation, but this is being super-cautious because patch.